### PR TITLE
Implement `CairoRunner::get_execution_resources`

### PR DESCRIPTION
**Pull request by azteca1998**
# Implement `CairoRunner::get_execution_resources`

## Description

Implementation `CairoRunner::get_execution_resources`.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Comment by azteca1998** [comment]
[add and export get_execution_resources() to PyCairoRunner](https://trello.com/c/vgWvsMdS/8-add-and-export-getexecutionresources-to-pycairorunner)

**Comment by azteca1998** [comment]
Note: depends on #501.

**Review by fmoletta — COMMENTED** [review]
Every PR should have tests for each added method/function.

**Review by juanbono — APPROVED** [review]

File: src/vm/runners/cairo_runner.rs [hunk 1]
```diff
@@ -443,6 +445,29 @@ impl CairoRunner {
...
     }
 
+    pub fn get_execution_resources(
+        &self,
+        vm: &VirtualMachine,
+    ) -> Result<ExecutionResources, TraceError> {
+        let n_steps = match self.original_steps {
+            Some(x) => x,
+            None => vm.trace.as_ref().ok_or(TraceError::TraceNotEnabled)?.len(),
+        };
```
Comment (Oppen) [inline src/vm/runners/cairo_runner.rs:455]: Is it OK for this to fail if both `vm.trace` and `self.original_steps` are `None`?

**Pull request merged**